task_sdk/src/airflow/sdk/definitions/asset/__init__.py

```diff
@@ -54,9 +55,26 @@ class AssetUniqueKey(NamedTuple):
     uri: str
 
     @staticmethod
-    def from_asset(asset: Asset) -> AssetUniqueKey:
+    def from_asset(asset: Asset | AssetModel) -> AssetUniqueKey:
         return AssetUniqueKey(name=asset.name, uri=asset.uri)
 
+    def to_asset(self) -> Asset:
+        return Asset(name=self.name, uri=self.uri)
+
+
+class AssetAliasUniqueKey(NamedTuple):
+    name: str
+
+    @staticmethod
+    def from_asset_alias(asset_alias: AssetAlias) -> AssetAliasUniqueKey:
+        return AssetAliasUniqueKey(name=asset_alias.name)
+
+    def to_asset_alias(self) -> AssetAlias:
+        return AssetAlias(name=self.name)
+
+
+BaseAssetUniqueKey = Union[AssetUniqueKey, AssetAliasUniqueKey]
```

Comment:
Just a by-the-way, not sure if you already know—these two classes aren’t actually distinctive due to how nametuples in Python are still fundamentally tuples. Specifically,

```pycon
>>> from collections import namedtuple
>>> A = namedtuple("A", "x y")
>>> B = namedtuple("B", "x y")
>>> A(1, 2) == B(1, 2)
True
```

The only reason this approach currently works is AssetUniqueKey and AssetAliasUniqueKey have different number of members (2 vs 1). They might as well be free functions returning tuples. Nothing we must change here though (it works after all); we can always fix this later if it fails to work in the future.

Reply:
Yep, this is what I expected (for easier serialization). But after a second thought, we probably could make it a class so that we can avoid the need to handle migration in the future (e.g., serialized named tuple key to something with identity check)

Reply:
changed it as an attrs class